src/OpenLoco/Windows/ToolTip.cpp

```diff
@@ -52,58 +60,28 @@ namespace OpenLoco::Ui::ToolTip
                 Ui::ToolTip::update((Ui::window*)regs.esi, regs.edx, regs.di, regs.ax, regs.bx);
                 return 0;
             });
-        registerHook(
-            0x004C9397,
-            [](registers& regs) FORCE_ALIGN_ARG_POINTER -> uint8_t {
-                draw((Ui::window*)regs.esi, (Gfx::drawpixelinfo_t*)regs.edi);
-                return 0;
-            });
-        registerHook(
-            0x004C94F7,
-            [](registers& regs) FORCE_ALIGN_ARG_POINTER -> uint8_t {
-                onClose((Ui::window*)regs.esi);
-                return 0;
-            });
-        registerHook(
-            0x004C94FF,
-            [](registers& regs) FORCE_ALIGN_ARG_POINTER -> uint8_t {
-                update((Ui::window*)regs.esi);
-                return 0;
-            });
     }
 
-    static void common(const window* window, int32_t widgetIndex, string_id stringId, int16_t cursorX, int16_t cursorY)
+    static void common(const window* window, int32_t widgetIndex, string_id stringId, int16_t cursorX, int16_t cursorY, FormatArguments& args)
     {
-        StringManager::formatString(byte_112CC04, stringId, _commonFormatArgs);
+        char buffer[512]{};
+        StringManager::formatString(buffer, stringId, &args);
 
         gCurrentFontSpriteBase = Font::medium_bold;
-        int16_t strWidth;
-        {
-            // gfx_get_string_width_new_lined
-            registers regs;
-            regs.esi = (uint32_t)&byte_112CC04[0];
-            call(0x00495715, regs);
-            strWidth = regs.cx;
-        }
+        int16_t strWidth = Gfx::getStringWidthNewLined(buffer);
         strWidth = std::max<int16_t>(strWidth, 196);
 
         gCurrentFontSpriteBase = Font::medium_bold;
-        {
-            // gfx_wrap_string
-            registers regs;
-            regs.esi = (uint32_t)&byte_112CC04[0];
-            regs.di = strWidth + 1;
-            call(0x00495301, regs);
-            strWidth = regs.cx;
-            _lineBreakCount = regs.di;
-        }
 
-        int width = strWidth + 3;
+        auto [wrappedWidth, breakCount] = Gfx::wrapString(buffer, strWidth + 1);
+        _lineBreakCount = breakCount;
+
+        int width = wrappedWidth + 3;
         int height = (_lineBreakCount + 1) * 10 + 4;
-        _widgets[0].right = width;
-        _widgets[0].bottom = height;
+        _widgets[widx::text].right = width;
+        _widgets[widx::text].bottom = height;
 
-        std::memcpy(&_text[0], &byte_112CC04[0], 512);
+        std::memcpy(&_text[0], buffer, 512);
```

Comment:
This looks like a workaround for the now-removed `loco_global`, which you did remove on line 193 below. Suggesting we do the same here:
```suggestion
        std::memcpy(_text, buffer, 512);
```

Reply:
In OpenRCT2 i just got rid of the buffer altogether. There isn't actually any need to for it we can write directly to _text. I'm going to make the same change here.